services/telemetry/telemetry.go

```diff
@@ -83,30 +83,34 @@ type Service struct {
 }
 
 // NewService creates a new service with given UUID and PMM version.
-func NewService(db *reform.DB, pmmVersion string) *Service {
+func NewService(db *reform.DB, pmmVersion string) (*Service, error) {
 	l := logrus.WithField("component", "telemetry")
 	s := &Service{
 		db:           db,
 		pmmVersion:   pmmVersion,
 		start:        time.Now(),
 		l:            l,
 		v1URL:        defaultV1URL,
-		v2Host:       defaultV2Host,
+		v2Host:       "",
 		interval:     defaultInterval,
 		retryBackoff: defaultRetryBackoff,
 		retryCount:   defaultRetryCount,
 	}
 
 	s.sDistributionMethod, s.tDistributionMethod, s.os = getDistributionMethodAndOS(l)
 
+	u, err := envvars.GetSAASHost(envV2Host)
+	if err != nil {
+		return s, err
+	}
+
+	s.v2Host = u
```

Comment:
It can be a little bit simpler.
```suggestion
	host, err := envvars.GetSAASHost(envV2Host)
	if err != nil {
		return s, err
	}

	s := &Service{
		db :        db,
		v2Host: host,
		....
```

Reply:
as I see `return s, err` will not work in this case

Reply:
Yes, it should be `return nil, err`. It's also common approach in go, to return empty results in case of error.

Reply:
Done